### PR TITLE
Merge search-by-location into Master

### PR DESCRIPTION
**Pull request by sehoven**
Wrote searching tests and search by location functionality.

Solves issues [#101](https://github.com/CMPUT301F16T01/Carrier/issues/101) and [#153](https://github.com/CMPUT301F16T01/Carrier/issues/153)

### CODE REVIEW

File: app/src/androidTest/java/comcmput301f16t01/github/carrier/SearchingTests.java
Comment (Hreherch) [inline app/src/androidTest/java/comcmput301f16t01/github/carrier/SearchingTests.java:36]: Is it working as expected?
Can you add a comment as to what your tearDown() method does?

File: app/src/main/java/comcmput301f16t01/github/carrier/Requests/ElasticRequestController.java
Comment (Hreherch) [inline app/src/main/java/comcmput301f16t01/github/carrier/Requests/ElasticRequestController.java]: Can we keep these somewhere else, like on the Wiki?

File: app/src/main/java/comcmput301f16t01/github/carrier/Requests/RequestController.java
Comment (Hreherch) [inline app/src/main/java/comcmput301f16t01/github/carrier/Requests/RequestController.java]: Did you complete this TODO while working on this branch? (I remember you mentioning it)

File: app/src/main/java/comcmput301f16t01/github/carrier/UserController.java
Comment (Hreherch) [inline app/src/main/java/comcmput301f16t01/github/carrier/UserController.java]: 💁 My bad, thank you for catching this!

File: app/src/androidTest/java/comcmput301f16t01/github/carrier/SearchingTests.java
Comment (Hreherch) [inline app/src/androidTest/java/comcmput301f16t01/github/carrier/SearchingTests.java:65]: In comments could I have what you are testing, and if you test for any exceptional cases? 
Also maybe include reference to which use case this tests, if applicable.

**Review by Hreherch — CHANGES_REQUESTED** [review]
I'm not sure if I can confirm/deny that the "null location" issue is correct, do you happen to have a test for it that I missed? (Not a necessary component to complete for approval) 

Otherwise doing great! Just some minor things to potentially change.

File: app/src/androidTest/java/comcmput301f16t01/github/carrier/SearchingTests.java
Comment (sehoven) [reply app/src/androidTest/java/comcmput301f16t01/github/carrier/SearchingTests.java:36]: Created an issue to implement the fix from the singleton-refactor branch once both are merged to master.

File: app/src/main/java/comcmput301f16t01/github/carrier/Requests/ElasticRequestController.java
Comment (sehoven) [reply app/src/main/java/comcmput301f16t01/github/carrier/Requests/ElasticRequestController.java]: Moved to Google Docs.

File: app/src/androidTest/java/comcmput301f16t01/github/carrier/SearchingTests.java
Comment (sehoven) [reply app/src/androidTest/java/comcmput301f16t01/github/carrier/SearchingTests.java:65]: Added use case references.

File: app/src/main/java/comcmput301f16t01/github/carrier/Requests/RequestController.java
Comment (sehoven) [reply app/src/main/java/comcmput301f16t01/github/carrier/Requests/RequestController.java]: Thank you. I removed this TODO. This is done.

**Comment by sehoven** [comment]
The null location fix is in SetLocationsActivity.java in the getCurrentLocation() method. I check if lastLocation is null (which only happens if there was a problem retrieving the current location). If it is null, it defaults to (0, 0) (which is somewhere in the middle of the ocean).

File: app/src/main/java/comcmput301f16t01/github/carrier/Requests/ElasticRequestController.java
Comment (Hreherch) [inline app/src/main/java/comcmput301f16t01/github/carrier/Requests/ElasticRequestController.java]: Git decided to be awkward.

Comment (Hreherch) [reply app/src/main/java/comcmput301f16t01/github/carrier/Requests/ElasticRequestController.java]: You don't have to fix this, it's in #167, I think. 
I'll approve this branch when you merge that with it's changes?

**Comment by Hreherch** [comment]
The only minor issue I can find is that I can offer twice on the same request if I haven't gone all the way back to the search buttons, this is more of an enhancement at this point. 
Also "Search Results s" is one of the titles. But that's more of a quality issue :P

**Review by Hreherch — APPROVED** [review]

**Pull request merged**
